platform/windows/display_server_windows.cpp

```diff
@@ -881,6 +894,16 @@ void DisplayServerWindows::_update_window_style(WindowID p_window, bool p_repain
 	DWORD style_ex = 0;
 
 	_get_window_style(p_window == MAIN_WINDOW_ID, wd.fullscreen, wd.borderless, wd.resizable, wd.maximized, wd.no_focus, style, style_ex);
+	if (wd.borderless) {
+		if (_try_enable_composition()) {
+			static const MARGINS shadows[2]{
+				{ 0, 0, 0, 0 },
+				// Add shadows to a borderless window
+				{ 1, 1, 1, 1 }
+			};
+			DwmExtendFrameIntoClientArea(wd.hWnd, shadows);
```

Comment:
This will cause conflicts with semi-transparent / non-rectangular windows, probably shadow should be a separate window flag (it is also possible to enabled/disable shadow on macOS, and it's disabled for the same reason).

Reply:
In terms of the shadow, I've spent ~2 hours trying to figure how to make a borderless window with shadows (this one is copied from a demo project, and is unofficial API, and it doesn't work without WS_THICKFRAME which messes with window rectangle). Stackoverflow also provides what seems like irrelevant information here.

https://github.com/melak47/BorderlessWindow/
https://stackoverflow.com/questions/43818022/borderless-window-with-drop-shadow

I don't think I will add shadow for borderless windows, as it would take way too much time to trial-and-error. If there is anybody with better Win32 experience than me they are welcomed to help out!

Reply:
Actually I managed to make shadows work. The Stackoverflow I found is just too old or something, just did a bit more google search and I found a solution that works:

https://stackoverflow.com/questions/60913399/border-less-winform-form-shadow/60916421#60916421